pkg/reconciler/v1alpha1/configuration/configuration.go

```diff
@@ -230,3 +253,59 @@ func (c *Reconciler) updateStatus(u *v1alpha1.Configuration) (*v1alpha1.Configur
 	}
 	return newu, nil
 }
+
+func (c *Reconciler) gcRevisions(ctx context.Context, config *v1alpha1.Configuration) error {
+	logger := logging.FromContext(ctx)
+
+	selector := labels.Set{serving.ConfigurationLabelKey: config.Name}.AsSelector()
+	revs, err := c.revisionLister.Revisions(config.Namespace).List(selector)
+	if err != nil {
+		return err
+	}
+
+	for _, rev := range revs {
+		gcConfig := configns.FromContext(ctx).RevisionGc
+
+		if isStale, err := isRevisionStale(gcConfig, rev, config.Spec.Generation); err != nil {
```

Comment:
If you just pass `ctx` into `isRevisionStale` then it can access the config directly **and** the logger, so that you don't have to access `lastPinStr` ignoring `, ok` below.